src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -171,18 +171,32 @@ private function dumpArrayList(string $output, string $input, string $contextPro
         $memberShape = $shape->getMember()->getShape();
 
         return strtr('
-            $index = -1;
-            foreach (INPUT as $listValue) {
-                $index++;
-                MEMBER_CODE
+            if (null !== INPUT) {
+                $index = -1;
+                INIT_CODE
```

Comment:
hmm.  This is alway `$payload = [];` right? no need for an other extra method

Reply:
I created the other method to ensure not doing this init everywhere as it break many tests.

Reply:
what if the test were wrong before?..

Reply:
Many are correct, for example. in DynamoDB, only List could be sent empty. Not Set (like SS, NS or BS).

Reply:
We could introduce a specific Shape like SetShape and keep ListShape for real list ? To allow making things differently